### ISSUE TITLE
Simplify the DAG-instantiation process

### ISSUE THREAD

**Issue by ghost** [astronomer/dag-factory#213]
Currently, once a `.yml` file has been created, the following code (or something like this) will have to be created to "instantiate" the DAG.

``` {python}
from dagfactory import DagFactory

# Pass in an exact config file name
dag_factory: DagFactory = DagFactory("/usr/local/airflow/dags/config/example_on_failure_callback.yml")

# Clean and generate DAGs
dag_factory.clean_dags(globals())
dag_factory.generate_dags(globals())
```

I think it would be more intuitive to do something like what is shown below. I know this is somewhat pedantic, and feel free to close this issue if so.

``` {python}
from dagfactory import DagFactory

# Pass in an exact config file name
dag_factory: DagFactory = DagFactory("/usr/local/airflow/dags/config/example_on_failure_callback.yml")
dag_factory.generate_dags()
```

**Comment by cmarteepants** [comment]
Yes! I agree - it feels like a lot of boilerplate as is, and IMO it's borderline dangerous if you forgot to clean the dags. Haven't tested but I can see there being a lot of confusion if, when hosting Airflow in K8s and using git-sync, a config file is deleted. The dag will still be there and will most likely be able to run. Then, if the dag processor container restarts (likely on k8s), the dag will "all of a sudden" go missing.

 @jroach-astronomer - is there any reason you can think of where you wouldn't want to remove no longer defined dags?

**Comment by jroach-astronomer** [comment]
Not one that I can think of, I'm right there with you, I think it would be dangerous to allow for no-longer-defined DAGs to exist.

**Comment by gyli** [comment]
I think we should consider even further simplification - creating DAG with YAML only and no Python script at all. A built-in "simple mode" can be added, that under a pre-defined folder, all YAML files will be scanned and applied. Some configuration of the simple mode can probably be done with env var if really necessary, but not with Python script.

**Comment by pankajastro** [comment]
Hi @ghost, We have simplified the interface in dag-factory 1.0.0. Regarding creating DAG only with YAML, the AIP-85 is the way forward.
